app/assets/javascripts/models/beta/iteration.js

```diff
@@ -1,21 +1,129 @@
-import moment from 'moment';
+import moment from "moment";
+import { status, storyTypes } from "libs/beta/constants";
+import * as Story from "./story";
 
 const weeksBetween = (dateA, dateB) =>
-  moment(new Date(dateA)).diff(moment(new Date(dateB)), 'week');
+  moment(new Date(dateA)).diff(moment(new Date(dateB)), "week");
 
 const getIterationForDate = (date, project) => {
   const weeks = weeksBetween(date, project.startDate);
   const iterationQuantity = Math.ceil(weeks / project.iterationLength);
 
-  return project.iterationLength > weeks ? iterationQuantity : iterationQuantity + 1;
+  return project.iterationLength > weeks
+    ? iterationQuantity
+    : iterationQuantity + 1;
 };
 
-export const getCurrentIteration = (project) => (
-  getIterationForDate(new Date(), project)
-);
+export const getDateForIterationNumber = (iterationNumber, project) => {
+  return moment(new Date(project.startDate))
+    .startOf("isoWeek")
+    .add(iterationNumber, "weeks")
+    .format("ddd MMM Do Y");
+};
+
+export const getCurrentIteration = project =>
+  getIterationForDate(new Date(), project);
 
 export const getIterationForStory = (story, project) => {
-  if (story.state === 'accepted') {
+  if (story.state === "accepted") {
     return getIterationForDate(new Date(story.acceptedAt), project);
   }
-}
+};
+
+const createSprint = (sprintNumber = 0, startDate = 0, isFiller = false) => ({
+  number: sprintNumber,
+  startDate: startDate,
+  points: 0,
+  completedPoints: null,
+  stories: [],
+  isFiller: isFiller
+});
+
+const createFillerSprints = (size, initialNumber, project) => {
+  let fillerSprints = [];
+  for (let i = 1; i <= size; i++) {
+    const sprintNumber = initialNumber + i;
+    const sprint = createSprint(
+      sprintNumber,
+      getDateForIterationNumber(sprintNumber, project),
+      true
+    );
+    fillerSprints.push(sprint);
+  }
+  return fillerSprints;
+};
+
+const lastRealSprintIndex = sprints => {
+  const realSprints =
+    sprints.length && sprints.filter(sprint => sprint && !sprint.isFiller);
+  const index = realSprints.length > 0 ? realSprints.length - 1 : 0;
```

Comment:
this can be improved like:

```
const index = realSprints.length ? realSprints.length - 1 : 0;
```

Reply:
I don't think so, when length === 0 it will return -1, because realSprints.length is defined. isn't that correct?

Reply:
No. 

```
const a = [];
a.length // equals 0
```

In this particular case, you can do something like:

```
const index = realSprints.length && realSprints.length - 1;
```

Reply:
Yes, you're right. Thanks!